### ISSUE TITLE
this is the case,

### ISSUE THREAD

**Issue by lixinsu** [linxiexiong/kbqa_bishe#1]
import torch
from torch.autograd import Variable
optimizer = torch.optim.Adagrad([X,Y],
                                    lr=100000)

X = Variable(torch.Tensor([1.2,1.3]),requires_grad=True)
Y = Variable(torch.Tensor([1.9,2.0]),requires_grad=True)
Z = X-Y
K = X + Y 
optimizer = torch.optim.Adagrad([X,Y],
                                    lr=100000)
loss_e = torch.mean(torch.clamp(-2 + K+ Z, min=0.0))
loss_e.backward()
optimizer.step()
print(Z.grad) # None 
print(X.grad) # 
"""
Variable containing:
 0
 0
[torch.FloatTensor of size 2]
"""

**Comment by lixinsu** [comment]
Autograd computes gradients for leaf nodes , 它不计算中间结果的grad ，为了节省空间，backward 里计算过程需要计算，但是后续过程不需要这些中间量的梯度。你所有网络层的参数都是叶子节点。

**Comment by lixinsu** [comment]
你可以看看你lstm里边linear层是否有梯度
